### PR TITLE
zehra Kopar Week1

### PR DESCRIPTION
**Pull request by zehrakopar**
JavaScript 3 week1 homework

### CODE REVIEW

File: homework/index.js [hunk 1]
```diff
@@ -30,13 +30,106 @@
...
+        createAndAppend('p', root, { text: err.message, class: 'alert' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        data.sort((a, b) => a.name.toLowerCase().localeCompare(b.name.toLowerCase()));
+        for (let i = 0; i < data.length; i++) {
+          createAndAppend('option', selector, { text: data[i].name, value: i });
+        }
+
+        const container = createAndAppend('div', root, { id: 'container' });
+        createAndAppend('div', container, { class: 'leftDiv' });
```
Comment (remarcmij) [inline homework/index.js]: It is a best practice to use kebab-case for CSS names: `leftDiv` -> `left-div`.

File: homework/index.js [hunk 2]
```diff
@@ -30,13 +30,106 @@
...
-      const root = document.getElementById('root');
+      const header = createAndAppend('header', root, { class: 'header' });
+      createAndAppend('p', header, { text: 'HYF Repositories' });
+      const selector = createAndAppend('select', header, { class: 'repo-selector' });
       if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+        createAndAppend('p', root, { text: err.message, class: 'alert' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        data.sort((a, b) => a.name.toLowerCase().localeCompare(b.name.toLowerCase()));
```
Comment (remarcmij) [inline homework/index.js]: Do not use `.toLowerCase()` here. All the logic, including taking into consideration the handling of 'case', is taken care of by `localeCompare`, which takes the language settings as the user has configured it in his/her computer. For example, I happen to know that there is no such thing as upper and lower case in Arabic.

File: homework/index.js [hunk 3]
```diff
@@ -30,13 +30,106 @@
...
+            class: 'contributor-item',
+            tabindex: '0',
+            'aria-label': contributor.login,
+          });
+
+          createAndAppend('img', li, {
+            src: contributor.avatar_url,
+            height: 48,
+            class: 'contributor-avatar',
+          });
```
Comment (remarcmij) [inline homework/index.js]: The HTML validator produces this error messages for the `img` tag:

```
Error: An “img” element must have an “alt” attribute, except under certain conditions. For details, consult guidance on providing text alternatives for images.
```

File: homework/index.js [hunk 4]
```diff
@@ -30,13 +30,106 @@
...
+    const rightDiv = document.querySelector('.rightDiv');
+    createAndAppend('p', rightDiv, {
+      class: 'contributor-header',
+      text: 'Contributions',
+    });
+    const ul = createAndAppend('ul', rightDiv, { class: 'contributor-list' });
+    fetchJSON(url, (err, contributors) => {
+      if (err) {
+        createAndAppend('p', root, { text: err.message, class: 'alert' });
+      } else {
```
Comment (remarcmij) [inline homework/index.js]: Upon detecting an error at the beginning of your callback function it is best to 'exit early' by returning after handling the error, as there is nothing useful left to do in the callback function. In this way you can get rid of the `else` clause and the indentation it requires.

```js
if (err) {
  createAndAppend('p', root, { text: err.message, class: 'alert' });
  return;
}

contributors.forEach(contributor => {
  ...
}

```

File: homework/index.js [hunk 5]
```diff
@@ -30,13 +30,106 @@
...
+            window.open(contributor.html_url);
+          });
+        });
+      }
+    });
+  }
+
   function main(url) {
+    const root = document.getElementById('root');
     fetchJSON(url, (err, data) => {
```
[the remark this answers is not in the document]
Comment (zehrakopar) [reply homework/index.js]: i ve got it. it will be done.

File: homework/style.css [hunk 6]
```diff
@@ -1,3 +1,128 @@
 .alert-error {
```
[the remark this answers is not in the document]
Comment (zehrakopar) [reply homework/style.css]: Actually i didn't look the sample application"s CSS. (except the box-shadow).

File: homework/index.js (hunk 1 above)
Comment (zehrakopar) [reply homework/index.js]: thank you..

File: homework/index.js (hunk 2 above)
Comment (zehrakopar) [reply homework/index.js]: got it..

File: homework/index.js (hunk 3 above)
Comment (zehrakopar) [reply homework/index.js]: Right, i will do.

File: homework/index.js (hunk 4 above)
Comment (zehrakopar) [reply homework/index.js]: Right, i will do.

**Comment by zehrakopar** [comment]
Yes Jim you are right. Last week was really difficult for me. I proceed to read some extra documents and watch videos about fetching data and Json. Also i ve asked my friends. i feel better now. i study for the promises too. thank you for your showing interest.

**Pull request closed without merging**